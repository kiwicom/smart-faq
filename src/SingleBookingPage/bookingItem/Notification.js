// @flow

import * as React from 'react';
import { Alert } from '@kiwicom/orbit-components';

import { formatCountDown } from '../../helpers/dateUtils';

type Props = {|
  isUrgent: boolean,
  hoursLeft: number,
|};

const Notification = ({ isUrgent, hoursLeft }: Props) => {
  const urgentMessage = `You depart in ${formatCountDown(hoursLeft)}.
  Don't hesitate to call us if you have an urgent problem.`;
  const normalMessage = `You depart in ${formatCountDown(hoursLeft)}.
  There is still time to add some nice extras or even change your booking.`;
  const type = isUrgent ? 'warning' : 'info';

  return (
    <div className="notification">
      <Alert type={type} icon>
        {isUrgent ? urgentMessage : normalMessage}
      </Alert>
      <style jsx>
        {`
          .notification {
            margin-bottom: 24px;
          }
        `}
      </style>
    </div>
  );
};

export const RawNotification = Notification;

export default Notification;
