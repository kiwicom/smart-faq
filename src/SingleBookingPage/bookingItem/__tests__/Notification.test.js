// @flow

import * as React from 'react';
import { render } from 'enzyme';
import ThemeProvider from '@kiwicom/orbit-components/lib/Theming/ThemeProvider';

import { RawNotification } from '../Notification';

describe('Notification', () => {
  it('should render', () => {
    const wrapper = render(
      <ThemeProvider>
        <RawNotification isUrgent hoursLeft={25} t={() => 'h'} />
      </ThemeProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should show warning when message is urgent', () => {
    const wrapper = render(
      <ThemeProvider>
        <RawNotification isUrgent hoursLeft={25} t={key => key} />
      </ThemeProvider>,
    );

    expect(wrapper.text()).toContain(
      "Don't hesitate to call us if you have an urgent problem.",
    );
  });

  it('should show just info when it is not urgent', () => {
    const wrapper = render(
      <ThemeProvider>
        <RawNotification isUrgent={false} hoursLeft={62} t={key => key} />
      </ThemeProvider>,
    );

    expect(wrapper.text()).toContain(
      'There is still time to add some nice extras or even change your booking.',
    );
  });
});
