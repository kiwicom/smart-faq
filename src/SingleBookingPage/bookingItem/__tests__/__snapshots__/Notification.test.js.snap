// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render 1`] = `
<div
  class="jsx-253481468 notification"
>
  <div
    class="Alert__StyledAlert-cJJYQC juWFnr"
  >
    <div
      class="Alert__IconContainer-eSlLEA eyJClo"
    >
      <svg
        fill="currentColor"
        height="24px"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align:middle"
        viewBox="0 0 24 24"
        width="24px"
      >
        <path
          d="M12 19.25a1.25 1.25 0 1 1 0-2.5 1.25 1.25 0 0 1 0 2.5zM11 10h2v6h-2v-6zm10.895 10.553l-9-18c-.339-.678-1.451-.678-1.79 0l-9 18c-.154.31-.138.678.044.973.183.294.504.474.851.474h18c.347 0 .668-.18.851-.474.182-.295.198-.663.044-.973z"
        />
      </svg>
    </div>
    <div
      class="Alert__ContentWrapper-glEZbs cPHqb"
    >
      <div
        class="Alert__Content-KRSaN flVfCN"
      >
        You depart in 25h.
  Don't hesitate to call us if you have an urgent problem.
      </div>
    </div>
  </div>
</div>
`;
